homeassistant/components/stiebel_eltron/climate.py

```diff
@@ -0,0 +1,153 @@
+    name = hass.data[STE_DOMAIN]['name']
+    ste_data = hass.data[STE_DOMAIN]['ste_data']
+
+    add_devices([StiebelEltron(name, ste_data)], True)
+    return True
+
+
+class StiebelEltron(ClimateDevice):
+    """Representation of a STIEBEL ELTRON heat pump."""
+
+    def __init__(self, name, ste_data):
+        """Initialize the unit."""
+        self._name = name
+
+        self._target_temperature = None
+        self._current_temperature = None
+        self._current_humidity = None
+        self._operation_modes = [STATE_AUTO, STATE_IDLE, STATE_DHW]
+        self._current_operation = None
+        self._filter_alarm = None
+        self._ste_data = ste_data
+
+        _LOGGER.debug("Initialized climate platform")
+
+    @property
+    def supported_features(self):
+        """Return the list of supported features."""
+        return SUPPORT_FLAGS
+
+    def update(self):
+        """Update unit attributes."""
+        self._ste_data.update()
+
+        self._target_temperature = self._ste_data.api.get_target_temp()
+        self._current_temperature = self._ste_data.api.get_current_temp()
+        self._current_humidity = self._ste_data.api.get_current_humidity()
+        self._filter_alarm = self._ste_data.api.get_filter_alarm_status()
+        self._current_operation = self._ste_data.api.get_operation()
+
+        _LOGGER.debug("Update %s, current temp: %s", self._name,
+                      self._current_temperature)
+
+    @property
+    def device_state_attributes(self):
+        """Return device specific state attributes."""
+        return {
+            ATTR_CURRENT_HUMIDITY: self._current_humidity,
+            'filter_alarm': self._filter_alarm
+        }
+
+    @property
+    def should_poll(self):
+        """Return the polling state."""
+        return True
+
+    @property
+    def name(self):
+        """Return the name of the climate device."""
+        return self._name
+
+    # Handle SUPPORT_TARGET_TEMPERATURE
+    @property
+    def temperature_unit(self):
+        """Return the unit of measurement."""
+        return TEMP_CELSIUS
+
+    @property
+    def current_temperature(self):
+        """Return the current temperature."""
+        return self._current_temperature
+
+    @property
+    def target_temperature(self):
+        """Return the temperature we try to reach."""
+        return self._target_temperature
+
+    @property
+    def target_temperature_step(self):
+        """Return the supported step of target temperature."""
+        return 0.1
+
+    @property
+    def min_temp(self):
+        """Return the minimum temperature."""
+        return 10.0
+
+    @property
+    def max_temp(self):
+        """Return the maximum temperature."""
+        return 30.0
+
+    def set_temperature(self, **kwargs):
+        """Set new target temperature."""
+        if kwargs.get(ATTR_TEMPERATURE) is not None:
+            self._target_temperature = kwargs.get(ATTR_TEMPERATURE)
```

Comment:
`update` will be called at the end of the service call after the actuator method call. So we don't need to update entity state holding instance attributes in the actuator methods.

Just save the temperature in a local variable. Do that first before checking for `None`.

Reply:
Accordingly modified

Reply:
I would like to revert your suggested changes. With not updating the entity state immediately, the UI does not react accordingly (due to the `@Throttle(MIN_TIME_BETWEEN_UPDATES)` in `__init__.py`) and it looks like selecting the operation mode did not work. Is this ok or do you have an alternative suggestion?

Reply:
Could we force an update overriding the throttle when updating after a service call? Would that be ok for the api? Otherwise we should revert.

A way to force an update could be to save a boolean as an instance attribute that we check in `update` and add another private update method, where we put the throttle instead, and call this update method from the main update method. If the boolean is true we force an update by passing `no_throttle=True` to the throttled update method.

Reply:
I could modify the call to the update method to do a forced update. It's not an issue for the api.